core/graph/src/main/java/uk/gov/gchq/gaffer/operation/export/graph/ExportToOtherGraph.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    public void setParentSchemaId(final String parentSchemaId) {
+        this.parentSchemaId = parentSchemaId;
+    }
+
+    public Schema getSchema() {
+        return schema;
+    }
+
+    public void setSchema(final Schema schema) {
+        this.schema = schema;
+    }
+
+    public String getGraphId() {
+        return graphId;
+    }
+
+    public void setGraphId(final String graphId) {
+        this.graphId = graphId;
+    }
+
+    public String getParentStorePropertiesId() {
+        return parentStorePropertiesId;
+    }
+
+    public void setParentStorePropertiesId(final String parentStorePropertiesId) {
+        this.parentStorePropertiesId = parentStorePropertiesId;
+    }
+
+    public StoreProperties getStoreProperties() {
+        return storeProperties;
+    }
+
+    public void setStoreProperties(final StoreProperties storeProperties) {
+        this.storeProperties = storeProperties;
+    }
+
+    public GraphLibrary getGraphLibrary() {
+        return graphLibrary;
+    }
+
+    public void setGraphLibrary(final GraphLibrary graphLibrary) {
+        this.graphLibrary = graphLibrary;
+    }
+
+    @Override
+    public TypeReference<T> getOutputTypeReference() {
+        return (TypeReference) new TypeReferenceImpl.Object();
+    }
+
+    @Override
+    public ValidationResult validate() {
+        final ValidationResult result = ExportTo.super.validate();
+
+        if (null == graphLibrary) {
+            // No graph library so we cannot look up the graphId/schemaId/storePropertiesId
+            if (null != parentSchemaId) {
+                result.addError("parentSchemaId cannot be used without a GraphLibrary");
+            }
+            if (null != parentStorePropertiesId) {
+                result.addError("parentStorePropertiesId cannot be used without a GraphLibrary");
+            }
+        } else if (graphLibrary.exists(graphId)) {
+            if (null != parentSchemaId) {
+                throw new IllegalArgumentException("GraphId " + graphId + " already exists so you cannot use a different schema. Do not set the parentSchemaId field");
+            }
+            if (null != schema) {
+                throw new IllegalArgumentException("GraphId " + graphId + "already exists so you cannot provide a different schema. Do not set the schema field.");
+            }
+            if (null != parentStorePropertiesId) {
+                throw new IllegalArgumentException("GraphId " + graphId + " already exists so you cannot use different store properties. Do not set the parentStorePropertiesId field");
+            }
+            if (null != storeProperties) {
+                throw new IllegalArgumentException("GraphId " + graphId + " already exists so you cannot provide different store properties. Do not set the storeProperties field.");
+            }
+        } else {
+            if (null != parentSchemaId) {
+                final Schema parentSchema = graphLibrary.getSchema(parentSchemaId);
```

Comment:
Would it be useful to have a method on the graph library API to check for the existence of a schema with a specified ID, e.g.

```
if (null != parentSchemaId && !graphLibrary.schemaExists(parentSchemaId))
    throw new IllegalArgumentException("Schema could not be found in the graphLibrary with id: " + parentSchemaId);
}
```

?